app/lib/actions/katello/repository/upload_files.rb

```diff
@@ -30,6 +33,7 @@ def plan(repository, files, content_type = nil)
 
             plan_action(FinishUpload, repository, content_type: content_type, upload_actions: upload_actions)
             plan_self(tmp_files: tmp_files)
+            plan_action(Actions::Katello::Applicability::Repository::Regenerate, :repo_ids => [repository.id]) if generate_applicability
           end
         ensure
```

Comment:
There's actually two ways to upload content.  The first takes the entire file and uploads it (i believe thats this action).  The problem with that is that only files of fairly small sizes are accepted.  This is what the UI uses.

The cli uses a much more complicated series of steps which breaks the file into chunks and uploads each of them and then finally triggers an import.  `app/lib/actions/katello/repository/import_upload.rb`  is what handles that. 

You can test this by running the 'hammer repository upload' command